File: IT/Input-Output/Esercizi/C++/esercizio9.cpp
/*
file esercizio9.cpp
compilare con: g++ -std=c++11 esercizio9.cpp
*/

#include <ctime>
#include <iostream>
#include <fstream>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::fstream;

int main()
{
	cout << "Apri il file: ";

	string filename;
	getline(cin, filename);

	fstream f(filename);

	if (f) {
		cout << "#### File esistente aperto in sola lettura. ####" << endl;
		string s;

		while (getline(f, s))
		{
			cout << s << endl;
		}
	}
	else {
		f.open(filename, fstream::out);
		time_t now = time(NULL);
		struct tm *timeinfo = localtime(&now);
		char buffer[80];
		strftime(buffer, sizeof(buffer), "File creato il %d-%m-%Y alle ore %H:%M:%S", timeinfo);

		f << string(buffer) << endl;
	}
}

File: IT/2/Input-Output/Esercizi/Python/esercizio9.py
# file esercizio9.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input
from os import path, linesep
from datetime import datetime

filename = input("Apri il file: ")

if path.isfile(filename):
	print("#### File esistente aperto in lettura. ####")
	f = open(filename, "r")
	for line in f:
		print(line, end="")
else:
	f = open(filename, "w")
	date = datetime.now()
	f.write("File creato il " + date.strftime("%d-%m-%Y alle ore %H:%M:%S") + linesep)

f.close()

